### ISSUE TITLE
Unable to scrape ScummVM games

### ISSUE THREAD

**Issue by richarnd** [muldjord/skyscraper#44]
I'm unable to scrape about half of my ScummVM games, and the half that I can scrape requires some hacking.

I have about 50 ScummVM games installed. A first scrape run gives no results (quits after 30 failures). Command entered is:

`Skyscraper --nobrackets --region us --videos -u xxx:yyy -s screenscraper -p scummvm`

(dropping the "region" or "-u" parameters has no effect)

If I rename all the ".svm" files to ".scummvm" and rerun the scrape, about half the games scrape correctly. This isn't ideal - ScummVM regenerates the .svm files when it starts, so the only way to avoid having duplicates in the list is to change the filenames back after the scrape, then manually edit the gamelist.xml.

I can get a few more scrapes by editing filenames further to match the filenames listed for each game on Screenscraper. This mostly involves removing platform-specific suffixes from the filenames (e.g. "kq2-amiga.scummvm" becomes "kq2.scummvm". It's not always required to do this. For example, "pq2-amiga.scummvm" scrapes successfully even though the only filename listed on Screenscraper is "pq2.scummvm". I'm not sure why this seems to matter sometimes, but sometimes does not. The fact that renaming from *.svm to *.scummvm suggests the filename on Screenscraper matters - they list only *.scummvm names.

In any case, about 25 games still won't scrape, despite the fact that their filenames precisely match those listed on Screenscraper. The following files don't scrape:

```
'baseball', No returned matches
'bc', No returned matches
'camelot', No returned matches
'dig', No returned matches
'eob', No returned matches
'eob2', No returned matches
'ft', No returned matches
'gk1', No returned matches
'iceman', No returned matches
'ihnm', No returned matches
'laurabow-amiga', No returned matches
'laurabow', No returned matches
'lol', No returned matches
'longbow', No returned matches
'loom-vga', No returned matches
'maniac', No returned matches
'scalpel', No returned matches
'monkey2', No returned matches
'scummvm', No returned matches
'simon1', No returned matches
'zak-v2', No returned matches
'simon2', No returned matches
'zak', No returned matches

```
These aren't rare games.

From reading threads on the Retropie forums, it sounds like Skyscraper is using more info than the filenames/checksums/hashes when scraping ScummVM games, possibly data from scummvm.ini? I tried editing some game names in that file (to match the name listed in Screenscraper), but that didn't help.

Running Retropie 4.4, Skyscraper v2.5.3. Internet connection is cable.

**Comment by muldjord** [comment]
Thank you for the detailed description! :)

Skyscraper scrapes scummvm games by looking them up in either scummvm.ini or ~/.scummrc (if it exists) and using the description name that matches with the square bracket name of the file. So if you have a file called "maniac.scummvm" it will look up a [maniac] section and use the name under description for that section while searching for the game. But NOT when using the 'screenscraper' module. For this module it sill uses the maniac.svm or maniac.scummvm names. So if that name exists in the screenscraper database, it will work (just tested it with maniac.svm and dig.scummvm).

Please update to the latest version of Skyscraper (Important, I've fixed something in the 'screenscraper' module recently), reset your scummvm.ini (important!) and all your filenames to match those of the original squarebracket names (and screenscraper filenames) for each game in '/opt/retropie/configs/scummvm/scummvm.ini' or '~/.scummvmrc' and try with:
```
$ Skyscraper -p scummvm -s thegamesdb --refresh
$ Skyscraper -p scummvm -s screenscraper --refresh
$ Skyscraper -p scummvm
```
The last command will scrape with all cached data. Let me know if that works.

**Comment by richarnd** [comment]
OK! Thanks for the quick response. If you'd prefer I post this on the Retropie forums where more will see it, let me know.

I restored the scummvm.ini from a backup I made before doing any editing. I then ran a ScummVM session to get it to regenerate the *.svm files, and deleted my *.scummvm files (the files should be identical, other than their extensions). I also updated Skyscraper from source to 2.5.7.

Short version is that this didn't work, or I'm missing a step.

```
$ Skyscraper -p scummvm -s thegamesdb --refresh
...

Elapsed time: 00:00:10
Estimated time left: 00:00:00

---- Scraping run completed! YAY! ----
Writing 557 (0 new) resources to local database, please wait... Success!
Now writing '/home/pi/RetroPie/roms/scummvm/gamelist.xml'... Success!!!

---- And here are some neat stats :) ----
Total completion time: 00:00:10

Average search match: 90%
Average entry completeness: 58%

Total number of games: 58
Successfully scraped games: 8
Skipped games: 50 (Filenames saved to '[homedir]/.skyscraper/skipped-thegamesdb.txt')
```


```
Skyscraper -p scummvm -s screenscraper --refresh 

Elapsed time: 00:00:23
Estimated time left: 00:00:22

This is NOT going well! I guit! *slams the door*
No, seriously, out of 30 files we had 30 misses. So either the scraping source is down or you are using a scraping source that doesn't support this platform. Please try another scraping module (check '--help').

Now exiting...
```

...i.e. thegamesdb search found 8/58 games (one of which is incorrect - it identified "scalpel.svm" as "Biing!: Sex, Intrigue and Scalpels", an "erotic hospital-management simulation" ;).

Screenscraper then found zero matches.

Should I change the file extension to *.scummvm again?

**Comment by muldjord** [comment]
Ok, so I just did som further testing. It seems it works well with 'screenscraper' but only if the file extensions are .scummvm. So please rename your svm files to scummvm files and try again.

You can easily rename them by running this command in the scummvm roms folder:
>$ rename 's/.svm/.scummvm/' *.svm

Then try again with:

>$ Skyscraper -p scummvm -s screenscraper --refresh

I just tested this with the following files, and it worked for all of them with Skyscraper 2.5.7:
>camelot.scummvm
>dig.scummvm
>eob2.scummvm
>eob.scummvm
>maniac.scummvm
>monkey2.scummvm
>simon1.scummvm

It won't work well with 'thegamesdb', so don't bother with that too much.

**Comment by richarnd** [comment]
I got it to scrape 100%, largely due to your help. The key step is making sure filenames match what's on screenscraper. Most obviously, that requires renaming *.svm to *.scummvm. But because in many cases only the primary (usually floppy DOS) version of scummvm games is listed on screenscraper, you also have to change the name of some games. For example, "kq2-amiga.scummvm" won't scrape, but "kq2.scummvm" will. You could just edit the filename, but this creates extra work because ScummVM auto-regenerates the *.svm files each time you run it based on the game's id in scummvm.ini. A better solution is to edit that file, or edit the Game ID within Scummvm. If you do that, then the autogenerated files have the filename you need. For example, change the ID of King's Quest 2/Amiga from "kq2-amiga" to "kq2". This also affects some (not all) CD versions of games, or games with "enhanced" versions, each of which ScummVM assigns a unique ID. If you change that ID to the "base" game, it will work.

Of course this solution won't work for completists that want to have multiple versions of the same game. I'm not sure what to do in that case.

Finally, you need to edit gamelist.xml, changing *.scummvm to *.svm. A global find & replace does the trick.

This is really the same thing I was doing before posting here, but either "--refresh" fixed something, or (more likely), upgrading to the new version of Skyscraper did.

It would be nice if all the versions of games (e.g. "kq2-amiga.scummvm") and the *.svm files Retropie generates were listed on screenscraper, but that's not your problem. I suppose you could code Skyscraper to search for "xyz.scummvm" when it sees "xyz.svm", but that might introduce unexpected behavior. On balance I'd say it's worth it but you might disagree.

**Comment by muldjord** [comment]
Thank you for the input. I am almost certain it was the update that fixed the issue as, pre-2.5.7, I didn't recognize the "ss" region that screenscraper uses for a LOT of games. I would assume especially scummvm games makes us of that generic region, as they aren't region coded.
